File: app/javascript/channels/web_presence_channel.js
import consumer from "channels/consumer"

window.webPresenceConsumer = consumer.subscriptions.create("WebPresenceChannel", {
  connected() {

  },

  disconnected() {
    this.perform('leave_link')
  },

  viewLink (linkId) {
    this.perform('view_link', {link_id: linkId})
  },

  leaveLink () {
    this.perform('leave_link')
  },

  received(data) {
  }
});

File: app/channels/web_presence_channel.rb
class WebPresenceChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
  end

  def unsubscribed
    leave_link
  end

  def view_link(data)
    if data['link_id']
      link = Link.find(data['link_id'])
      if link && connection.current_user
        connection.current_user.view_link(link)
      end
    end
  end

  def leave_link
    if connection.current_user
      connection.current_user.leave_link
    end
  end
end
